### PR TITLE
fixup! (1.19 backport) [FLINK-35886][task] Fix watermark idleness timeout accounting when subtask is backpressured/blocked

### PR DESCRIPTION
**Pull request by pnowojski**
this fixes https://github.com/apache/flink/pull/25223 due to https://github.com/apache/flink/pull/25223#discussion_r1728928197

### CODE REVIEW

**Review by rkhachatryan — APPROVED** [review]
I'm fine with reverting, but we don't really have to.
Instead, we can add the `default` implementation as suggested [here](https://github.com/apache/flink/pull/25223#discussion_r1723771139).

Older code won't have to be re-compiled according to https://docs.oracle.com/javase/tutorial/java/IandI/defaultmethods.html#:~:text=Default%20methods%20enable%20you%20to%20add%20new%20functionality%20to%20existing,as%20parameters%20to%20existing%20interfaces. (I also quickly checked that locally).

File: pom.xml [hunk 1]
```diff
@@ -2355,8 +2355,6 @@ under the License.
 								<exclude>@org.apache.flink.annotation.Experimental</exclude>
 								<exclude>@org.apache.flink.annotation.Internal</exclude>
 								<!-- MARKER: start exclusions; these will be wiped by tools/releasing/update_japicmp_configuration.sh -->
-								<!-- New method has been added to the interface, but this interface shouldn't be implemented by users, only used/called, so this doesn't break API compatibility. -->
-								<exclude>org.apache.flink.api.common.eventtime.WatermarkGeneratorSupplier$Context</exclude>
 								<!-- Base interface has been extracted from the existing Clock class. This interface shouldn't affect user code. -->
 								<exclude>org.apache.flink.util.clock.Clock</exclude>
```
Comment (pnowojski) [inline pom.xml:2359]: I'm keeping this incompatibility excluded as is. I think/hope it should be fine.

**Comment by pnowojski** [comment]
Thanks @rkhachatryan for looking into this. Let's try without reverting.

**Review by rkhachatryan — APPROVED** [review]
Thanks for updating the PR, LGTM

File: pom.xml (hunk 1 above)
Comment (afedulov) [reply pom.xml:2361]: @pnowojski unfortunately this breaks the japicmp check:
```
[ERROR] Failed to execute goal com.github.siom79.japicmp:japicmp-maven-plugin:0.17.1:cmp (default) on project flink-core: There is at least one incompatibility: org.apache.flink.api.common.eventtime.WatermarkGeneratorSupplier$Context.getInputActivityClock():METHOD_NEW_DEFAULT,org.apache.flink.util.clock.Clock:METHOD_ABSTRACT_ADDED_IN_IMPLEMENTED_INTERFACE -> [Help 1]
```
This currently presents a blocker for 1.20.1 and 1.19.2 releases. Could you please look into providing a fix?

**Pull request merged**